net

```diff
@@ -1113,21 +1113,23 @@ def do_connect(conn, psk=None):
     set_mac(iface, mac)
     set_dns(dns)
 
-    if not wpa:
-        if is_wifi_iface(iface):
-            print 'Connecting to %s' % ssid
-            cmd = 'iwconfig %s key open ap %s' % (iface, ap_addr or 'any')
-            if ssid:
-                escaped_ssid = ''.join("\\x%02x" % ord(c) for c in ssid)
-                cmd += ' essid -- "%s"' % escaped_ssid
-            else:
-                cmd += ' essid any'
-            run(cmd)
+    if not wpa and not is_wifi_iface(iface):
         if not addr:
             print 'Sending DHCP request'
         for cmd in network:
             run_or_die(cmd)
     else:
+        if not wpa:
```

Comment:
else:
    if ...:

Why not

elif ...:

?

Reply:
Falling through to wpa.
The logic overall is shitty--not only in regards to else/elif.

Reply:
Added clarifying comments in https://github.com/br0ns/net/pull/10/commits/d49dede98c41ede2cc25502a9d0d1c53f2088b07 .